### PR TITLE
Update unit test docs in compliance with testing-library release

### PR DESCRIPTION
**Pull request by jophy-ye**
### What
Updated slightly out-of-date docs of Vitest & Jest Manual Setup to install required peer dependency to work with React Testing Library v16, which was released 3 weeks ago at the time of writing.

### Why
Beginning from [React Testing Library v16](https://github.com/testing-library/react-testing-library/releases/tag/v16.0.0), `@testing-library/dom` has become a peer dependency and needs to be explicitly installed alongside `@testing-library/react`.

### How
Install the package in the example command.

### CODE REVIEW

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 7304e01c01dcea715e607af0ea70c1c54db3aac0

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Review by eps1lon — APPROVED** [review]
Thank you

**Comment by jophy-ye** [comment]
Hmm... I wonder why some of the build tests are not successful when I updated several lines of the docs. In fact, it seems the last 10 pull requests with the "Documentation" tag do not pass most tests. Interesting🤔

File: docs/02-app/01-building-your-application/08-testing/01-vitest.mdx [hunk 1]
```diff
@@ -21,15 +21,17 @@ npx create-next-app@latest --example with-vitest with-vitest-app
...
+yarn add -D vitest @vitejs/plugin-react jsdom @testing-library/react @testing-library/dom
 # or
-pnpm install -D vitest @vitejs/plugin-react jsdom @testing-library/react
+pnpm install -D vitest @vitejs/plugin-react jsdom @testing-library/react @testing-library/dom
 # or
-bun add -D vitest @vitejs/plugin-react jsdom @testing-library/react
+bun add -D vitest @vitejs/plugin-react jsdom @testing-library/react @testing-library/dom
 ```
 
+Beginning from [React Testing Library 16](https://github.com/testing-library/react-testing-library/releases/tag/v16.0.0), `@testing-library/dom` is a peer dependency and needs to be explicitly installed.
```
Comment (devjiwonchoi) [inline docs/02-app/01-building-your-application/08-testing/01-vitest.mdx:33]: For further readers, I'm expecting them to be installing above 16, do we still need this explanation?

**Comment by jophy-ye** [comment]
@devjiwonchoi You have your point. Indeed, this page is more of a tutorial than a formal API reference, intending to help people setup a minimum viable example.

**Comment by devjiwonchoi** [comment]
Cool! Can you remove it? We can merge afterwards 🙏

**Comment by jophy-ye** [comment]
I’m extremely sorry if this question seems naive: should I commit the deletion and preserve history, or change the existing commit to make the history merged into next.js cleaner? I can do it right away. Thank you so much.

**Comment by devjiwonchoi** [comment]
No need to apologize, I'm here to help! 😆

You can remove those explanation and push it to the current branch!

File: docs/02-app/01-building-your-application/08-testing/01-vitest.mdx [hunk 2]
```diff
@@ -21,15 +21,17 @@ npx create-next-app@latest --example with-vitest with-vitest-app
...
+
```
Comment (eps1lon) [inline docs/02-app/01-building-your-application/08-testing/01-vitest.mdx:34]: Removed as per https://github.com/vercel/next.js/pull/67217/files#r1656280905
```suggestion
```

File: docs/02-app/01-building-your-application/08-testing/02-jest.mdx [hunk 3]
```diff
@@ -23,13 +23,15 @@ Since the release of [Next.js 12](https://nextjs.org/blog/next-12), Next.js now
...
 # or
-yarn add -D jest jest-environment-jsdom @testing-library/react @testing-library/jest-dom
+yarn add -D jest jest-environment-jsdom @testing-library/react @testing-library/dom @testing-library/jest-dom
 # or
-pnpm install -D jest jest-environment-jsdom @testing-library/react @testing-library/jest-dom
+pnpm install -D jest jest-environment-jsdom @testing-library/react @testing-library/dom @testing-library/jest-dom
 ```
 
+Beginning from [React Testing Library 16](https://github.com/testing-library/react-testing-library/releases/tag/v16.0.0), `@testing-library/dom` is a peer dependency and needs to be explicitly installed.
+
```
Comment (eps1lon) [inline docs/02-app/01-building-your-application/08-testing/02-jest.mdx:34]:
```suggestion
```

File: docs/02-app/01-building-your-application/08-testing/01-vitest.mdx (hunk 1 above)
Comment (eps1lon) [reply docs/02-app/01-building-your-application/08-testing/01-vitest.mdx]: Done

**Pull request merged**